Source/Scene/GroundPrimitive.js

```diff
@@ -0,0 +1,655 @@
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsStencilDepthPass,
+                    shaderProgram : this._sp,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+
+                this._colorPassCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsColorPass,
+                    shaderProgram : this._sp,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+
+                this._pickCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsPickPass,
+                    shaderProgram : this._spPick,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+            }
+        }
+
+        var stencilPreloadCommands = this._stencilPreloadPassCommands;
+        var stencilDepthPassCommands = this._stencilDepthPassCommands;
+        var colorPassCommands = this._colorPassCommands;
+        var pickCommands = this._pickCommands;
+
+        var j;
+        var length = primitiveCommandList.length;
+        for (j = 0; j < length; ++j) {
+            var command = primitiveCommandList[j];
+
+            var stencilPreloadCommand = stencilPreloadCommands[j];
+            stencilPreloadCommand.boundingVolume = command.boundingVolume;
+
+            var stencilDepthPassCommand = stencilDepthPassCommands[j];
+            stencilDepthPassCommand.boundingVolume = command.boundingVolume;
+
+            var colorCommand = colorPassCommands[j];
+            colorCommand.boundingVolume = command.boundingVolume;
+            colorCommand.debugShowBoundingVolume = this.debugShowBoundingVolume;
+
+            var pickCommand = pickCommands[j];
+            pickCommand.boundingVolume = pickCommand.boundingVolume;
+        }
+
+        var passes = frameState.passes;
+
+        if (passes.render) {
+            for (j = 0; j < length; ++j) {
+                commandList.push(stencilPreloadCommands[j], stencilDepthPassCommands[j], colorPassCommands[j]);
```

Comment:
Any idea how the performance of this is compared to three calls to `push`?  @shunter?